dbms/src/Interpreters/Join.cpp

```diff
@@ -1304,6 +1304,9 @@ Block Join::doJoinBlockHash(ProbeProcessInfo & probe_process_info) const
                 offsets_to_replicate->assign(
                     offsets_to_replicate->begin() + probe_process_info.start_row,
                     offsets_to_replicate->begin() + probe_process_info.end_row);
+                if (isAntiJoin(kind) && filter != nullptr)
+                    filter->assign(filter->begin() + probe_process_info.start_row,
```

Comment:
Better not to use `assign` because it uses `memcpy` which has an undefined behavior if two memory locations are overlapped.

Reply:
I aggree, but since it is close to the release deadline, I suggest we keep `assign` in this pr and fix it later since `assign` is used many places, it will bring extra risk if we change too many in this pr.